ext/imap/php_imap.c

```diff
@@ -4316,101 +4311,157 @@ static void _php_make_header_object(zval *myzvalue, ENVELOPE *en)
 	}
 
 	if (en->to) {
```

Comment:
So the code changes here look correct. But I also think there's more opportunity to refactor here. It's always the same control structure but the only thing that differs is which addresses are passed in (e.g. en->to, en->bcc, ...) and the key name (e.g. "toaddress", ...).
I don't know, it's just an idea...